src/com/dotcms/api/system/user/UserServiceFactory.java

```diff
@@ -0,0 +1,366 @@
+				this(inode, permissionType, filter, start, limit, false, true);
+			}
+
+			/**
+			 * Creates an instance of this class.
+			 * 
+			 * @param inode
+			 *            - The Inode of a given asset in order to get the list
+			 *            of users that have access to it.
+			 * @param permissionType
+			 *            - The permission type that users with access to the
+			 *            asset Inode must have.
+			 * @param filter
+			 *            - The string or characters that are part of the first
+			 *            name, last name, or e-mail of the user.
+			 * @param start
+			 *            - For pagination purposes. Lower range of the set to
+			 *            include in the query result.
+			 * @param limit
+			 *            - For pagination purposes. Upper range of the set to
+			 *            include in the query result.
+			 * @param includeAnonymous
+			 *            - If set to {@code true}, the "Anonymous" user will be
+			 *            included in the result. Otherwise, set to
+			 *            {@code false}.
+			 * @param includeDefault
+			 *            - If set to {@code true}, the "Default" user will be
+			 *            included in the result. Otherwise, set to
+			 *            {@code false}.
+			 */
+			public UsersListTemplate(String inode, int permissionType, String filter, int start, int limit,
+					boolean includeAnonymous, boolean includeDefault) {
+				this.inode = inode;
+				this.permissionType = permissionType;
+				this.filter = filter;
+				this.start = start;
+				this.limit = limit;
+				this.includeAnonymous = includeAnonymous;
+				this.includeDefault = includeDefault;
+			}
+
+			/**
+			 * Executes this filtering template based on the specified search
+			 * criteria.
+			 * 
+			 * @return A {@link Map} with the result of the filtering query.
+			 */
+			public Map<String, Object> perform() {
+				ArrayList<Map<String, String>> list = null;
+				Map<String, Object> results = new HashMap<String, Object>(2);
+				int totalItemCount = 0;
+				List<User> users = null;
+				int realUserCount = 0;
+				try {
+					// Step 1. Retrieve users, beginning from "start" parameter,
+					// up to a number of "limit" items, filtered by "filter"
+					// parameter.
+					totalItemCount = getUserCount();
+					if (start < totalItemCount) {
+						users = getUsers();
+						realUserCount = users.size();
+					}
+					// Step 2. Assemble all of this information into an
+					// appropriate container to the view
+					if (users != null) {
+						int pageSize = realUserCount;
+						list = new ArrayList<Map<String, String>>(pageSize);
+						for (User aUser : users) {
+							Map<String, String> aRecord = new HashMap<String, String>();
+							String fullName = aUser.getFullName();
+							fullName = (UtilMethods.isSet(fullName) ? fullName : " ");
+							String emailAddress = aUser.getEmailAddress();
+							emailAddress = (UtilMethods.isSet(emailAddress) ? emailAddress : " ");
+							aRecord.put("id", aUser.getUserId());
```

Comment:
You can reduce this to map("id", aUser.getUserId(), "type", USER_TYPE_VALUE ....)
from CollectionUtils.map